python/doc/source/extending.rst

```diff
@@ -340,10 +342,20 @@ To build this module, you will need a slightly customized ``setup.py`` file
         ext.include_dirs.append(pa.get_include())
         ext.libraries.extend(pa.get_libraries())
         ext.library_dirs.extend(pa.get_library_dirs())
-        # Try uncommenting the following line on Linux if you otherwise
-        # get weird linker errors or runtime crashes
-        #ext.define_macros.append(("_GLIBCXX_USE_CXX11_ABI", "0"))
 
-    setup(
-        ext_modules=ext_modules,
-    )
+        if os.name == 'posix':
+            ext.extra_compile_args.append('-std=c++11')
+
+        # Try uncommenting the following line on Linux
+        # if you get weird linker errors or runtime crashes
+        # ext.define_macros.append(("_GLIBCXX_USE_CXX11_ABI", "0"))
+
+
+    setup(ext_modules=ext_modules)
```

Comment:
Doesn't have to be done in this patch but i wonder if this example can be included from a .py file so it can be tested more easily